backend/canisters/user_index/impl/src/lifecycle/post_upgrade.rs

```diff
@@ -24,4 +24,15 @@ fn post_upgrade(args: Args) {
     init_state(env, data, args.wasm_version);
 
     info!(version = %args.wasm_version, "Post-upgrade complete");
+
+    mutate_state(|state| {
```

Comment:
Perhaps the canister should remember if it has synced already so it doesn't try again (if we forget to remove the code)

And perhaps we should label the code to be removed